src/main/java/io/nats/client/Options.java

```diff
@@ -1234,7 +1270,14 @@ public String buildProtocolConnectOptionsString(String serverURI, boolean includ
         appendOption(connectString, Options.OPTION_TLS_REQUIRED, String.valueOf(this.isTLSRequired()), false, true);
         appendOption(connectString, Options.OPTION_ECHO, String.valueOf(!this.isNoEcho()), false, true);
 
-        if (includeAuth) {
+        if (includeAuth && nonce != null && this.getAuthHandler() != null) {
+            String nkey = this.getAuthHandler().getID();
+            byte[] sig = this.getAuthHandler().sign(nonce);
+            String encodedSig = Base64.getEncoder().encodeToString(sig);
+
+            appendOption(connectString, Options.OPTION_NKEY, nkey, true, true);
+            appendOption(connectString, Options.OPTION_SIG, encodedSig, true, true);
+        } else if (includeAuth) {

```

Comment:
In Go client, when processing the info, if we get a nonce (that is, from a server that supports it), and the NKey option is not set (in your case, a auth handler) then we fail:
```
func (nc *Conn) processExpectedInfo() error {
(...)
	if nc.Opts.Nkey != "" && nc.info.Nonce == "" {
		return ErrNkeysNoSupported
	}
(...)
```
I am not seeing that in this file. Did I miss it?